In [1]:
import tflearn
from tflearn.data_utils import to_categorical, pad_sequences
from tflearn.datasets import imdb

### imdb train/test split for 10000 line items

In [2]:
train, test, _ = imdb.load_data(path = "imdb.pkl", n_words = 10000, valid_portion=0.1)
trainX, trainY = train
testX, testY = test

In [12]:
print("Type of Train X & Y:")
print(type(trainX))
print(type(trainY))
print("Train X[0] :")
print (trainX[0])
print("Train Y[0] :")
print (trainY[0])
print("Train X[0] :")
print (trainX[1:10])
print("Train Y[0] :")
print (trainY[1:10])

Type of Train X & Y:
<class 'list'>
<class 'list'>
Train X[0] :
[17, 25, 10, 406, 26, 14, 56, 61, 62, 323, 4]
Train Y[0] :
0
Train X[0] :
[[16, 586, 32, 885, 17, 39, 68, 31, 2994, 2389, 328, 4], [1, 2, 1, 139, 6, 130, 1, 5, 6, 25, 105, 4730, 40], [6691, 1, 10, 333, 10, 17, 27, 4, 34, 181, 6, 1418, 256, 4], [30, 287, 142, 2216, 707, 3763, 20, 68, 57, 30, 37, 309, 14, 4], [224, 3, 371, 3, 1, 4, 128, 37, 16, 90, 48, 1247, 8, 79, 294, 913, 1709, 4], [17, 10, 2, 6409, 25, 325, 7, 3161, 4, 2588, 1977, 176, 3, 26, 50, 35, 70, 1050, 395, 4], [16, 104, 32, 138, 152, 16, 49, 17, 25, 48, 89, 3, 26, 16, 128, 91, 1445, 7, 164, 14, 4], [70, 922, 87, 394, 25, 3, 129, 883, 53, 457, 86, 879, 87, 70, 297, 42, 41, 86, 1752, 14, 40], [2, 118, 35, 6186, 5, 2, 242, 10, 397, 4, 14, 20, 6, 456, 7, 2, 1938, 7, 1, 5, 1, 4]]
Train Y[0] :
[0, 0, 1, 0, 0, 1, 1, 0, 0]


#### 16, 586 represents the number of occurance of the words in pickled index
#### 0,0,1 represents the sentiments

# Data preprocessing
### Sequence padding

In [14]:
trainX = pad_sequences(trainX, maxlen=100, value = 0.)
testX = pad_sequences(testX, maxlen=100, value = 0.)

#Convert to Binary vectors
trainY = to_categorical(trainY, nb_classes=0.)
testY = to_categorical(testY, nb_classes=0.)

## Building Neural Network

In [15]:
net = tflearn.input_data([None, 100])
net = tflearn.embedding(net, input_dim=10000, output_dim=128)
net = tflearn.lstm(net, 128, dropout=0.8)
net = tflearn.fully_connected(net, 2, activation='softmax')
net = tflearn.regression(net, optimizer='adam', learning_rate=0.001, loss='categorical_crossentropy')


### Training the model

In [17]:
model = tflearn.DNN(net, tensorboard_verbose=0)
model.fit(trainX, trainY, validation_set =(testX, testY), show_metric = True, batch_size = 32)

Training Step: 7039  | time: 250.862s
| Adam | epoch: 010 | loss: 0.00000 - acc: 0.9819 -- iter: 22496/22500
Training Step: 7040  | time: 260.441s
| Adam | epoch: 010 | loss: 0.00000 - acc: 0.9837 | val_loss: 0.67210 - val_acc: 0.8028 -- iter: 22500/22500
--
